### PR TITLE
smartsynchronize: Fix checkver and force update to 4.0.0

### PR DESCRIPTION
**Pull request by niheaven**

### CODE REVIEW

File: bucket/smartsynchronize.json [hunk 1]
```diff
@@ -27,16 +27,16 @@
     },
     "persist": ".settings",
     "checkver": {
-        "url": "https://www.syntevo.com/smartsynchronize/changelog.txt",
-        "regex": "([\\d.]+)"
+        "url": "https://www.syntevo.com/smartsynchronize/download/",
+        "regex": "smartsynchronize-portable-([\\d.]+).7z"
```
Comment (Ash258) [inline bucket/smartsynchronize.json]:
```suggestion
        "regex": "smartsynchronize-portable-([\\d.]+)\\.7z"
```

Comment (niheaven) [reply bucket/smartsynchronize.json]: In this situation, `.` is enough, there should be one character `.`, so regex `.` will match it. Of couse, `\.` could be used, too.

Comment (Ash258) [reply bucket/smartsynchronize.json]: There is no need to expect anything when you know there will always be dot literally.

**Comment by r15ch13** [comment]
`(?sm)` is another neat regex trick I didn't know about. Thanks :smile:

**Pull request merged**